### PR TITLE
refactor: avoid passing duplicated mainfest to insights.collect()

### PR DESCRIPTION
**Pull request by xiangce**
- the "client_config" (InsightsConfig) already contains the
  manifest read from the client tool, this change removes it
  from the argument list of insights.collect() and simplify
  the code

### All Pull Requests:

Check all that apply:

* [ ] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### CODE REVIEW

File: insights/collect.py [hunk 1]
```diff
@@ -403,35 +403,38 @@ def generate_archive_name():
...
         (str, dict): The full path to the created tar.gz or workspace.
         And a dictionary with relevant exceptions captured by the broker during
         core collection, this dictionary has the following structure:
         ``{ exception_type: [ (exception_obj, component), (exception_obj, component) ]}``.
     """
-
+    # Get the manifest from client configuration
+    manifest = default_manifest
+    if hasattr(client_config, 'manifest') and client_config.manifest:
+        manifest = client_config.manifest
```
Comment (JoySnow) [inline insights/collect.py:435]: @xiangce   The manifest file load part from the origin code in [insights/client/core_collector.py  Line#35](https://github.com/RedHatInsights/insights-core/pull/4058/files#diff-76352cb529a81903277207d22b270de96a23532f285893c7f37e697c6f5109afL35) is skipped here. 

```
         if self.config.app is None:
                with open(self.config.manifest, 'r') as f:
                    manifest = f.read()
```

I see the `load_manifest(manifest)` calling below doesn't do the file load either. 

Then how do we load manifest from file?  Or we don't need the file load?

Comment (xiangce) [reply insights/collect.py:435]: The `load_mainfest` is strong enough to handle both `dict` and `file`:
https://github.com/RedHatInsights/insights-core/blob/729a1ca28e25468f60abd707bcaccb7b75bcb300/insights/collect.py#L280-L287

Comment (xiangce) [reply insights/collect.py:435]: Or, we should improve it to hand a `path` to a file.

Comment (JoySnow) [reply insights/collect.py:435]: Yep, we still need to do the `open(mainfest, 'r')` for `load_mainfest`.

**Review by JoySnow — APPROVED** [review]
@xiangce  The changes LGTM.

**Pull request merged**